testkit/src/main/java/org/apache/calcite/test/catalog/MockCatalogReader.java

```diff
@@ -618,6 +621,57 @@ public StructKind getKind() {
     }
   }
 
+  /**
+   * Mock implementation of
+   * {@link org.apache.calcite.prepare.Prepare.PreparingTable} which supplies MaxRow stat.
+   */
+  public static class MaxRowMockTable
+      extends MockTable implements BuiltInMetadata.MaxRowCount.Handler {
```

Comment:
Maybe you can avoid introducing a new sub-class and make `MockTable` implement the `MaxRowCount.Handler` assuming that rowCount == maxRowCount at this point. If we need to differentiate in the future we can add or modify the constructors/factories in `MockTable`.